examples/token_spec.rs

```diff
@@ -4,124 +4,120 @@ use std::hash::{Hash, Hasher};
 
 #[macro_use] extern crate prattle;
 
-use prattle::node::Node;
-use prattle::spec::ParserSpec;
-use prattle::lexer::LexerVec;
-use prattle::parser::{GeneralParser, Parser};
-use prattle::precedence::PrecedenceLevel;
+use prattle::prelude::*;
 
 #[derive(Debug, Clone, Eq, Ord)]
-pub enum Token {
+pub enum CToken {
     Number(String), 
     Ident(String), 
     Add, Sub, 
     Mul, Div, Mod, 
     LParens, RParens
 }
 
-impl PartialEq for Token {
-    fn eq(&self, other: &Token) -> bool {
+impl PartialEq for CToken {
+    fn eq(&self, other: &CToken) -> bool {
         let lhs = match self {
-            Token::Number(_) => 0, 
-            Token::Ident(_) => 1, 
-            Token::Add => 2, 
-            Token::Sub => 3,
-            Token::Mul => 4,
-            Token::Div => 5,
-            Token::Mod => 6,
-            Token::LParens => 7,
-            Token::RParens => 8,
+            CToken::Number(_) => 0, 
+            CToken::Ident(_) => 1, 
+            CToken::Add => 2, 
+            CToken::Sub => 3,
+            CToken::Mul => 4,
+            CToken::Div => 5,
+            CToken::Mod => 6,
+            CToken::LParens => 7,
+            CToken::RParens => 8,
         };
         let rhs = match other {
-            Token::Number(_) => 0, 
-            Token::Ident(_) => 1, 
-            Token::Add => 2, 
-            Token::Sub => 3,
-            Token::Mul => 4,
-            Token::Div => 5,
-            Token::Mod => 6,
-            Token::LParens => 7,
-            Token::RParens => 8,
+            CToken::Number(_) => 0, 
+            CToken::Ident(_) => 1, 
+            CToken::Add => 2, 
+            CToken::Sub => 3,
+            CToken::Mul => 4,
+            CToken::Div => 5,
+            CToken::Mod => 6,
+            CToken::LParens => 7,
+            CToken::RParens => 8,
         };
         lhs.eq(&rhs)
     }
 }
```

Comment:
https://doc.rust-lang.org/std/mem/fn.discriminant.html

Reply:
Actually all this stuff seems bad. I wouldn't expect that `CToken::Number(0) == CToken::Number(1)`. How about
```rust
trait Token {
    fn getClass(&self) -> usize;
}
```

 or something? Then you can use `#[derive]`.

Reply:
It's needed for the HashMap in ParserSpec, I could use that discriminant function to skip around that.

Reply:
Checked and confirmed it works correctly, removed dependency on Hash impl behaving like this. Now can just do discriminant(&token). Thank you!